### DISCUSSION TITLE
Solax PV no longer shows up in the device list.

### DISCUSSION THREAD

**Discussion by UdoD5** [freakent/dbus-mqtt-devices#105] [General]
### Version

0.6.5 (current)

### What version of VenusOS are you running on your GX device

3.21 (current release)

### Does this issue occur during installation or at runtime ?

Runtime

### What is the issue you are actually experiencing?

Hi, 
I'm using a ESP device to read the values from my solax X1 mini PV inverter and provide the values via dbus mqtt to the Cerbo GX. 
Unfortunately with the change to Firmware 3.21 the connection broke (upating from 3.14). The entry was still there but no values where received. I switched back to 3.14 and it worked again. So I assumed I probably need update the dbus-mqtt-devices to version 0.6.5. 

This is the MQTT code from the ESPHome:

mqtt:
  broker: 192.168.1.61 # +++  MQTT-Server
  port: 1883
  discovery: false
  birth_message:
    topic: device/${client_id}/Status
    payload: '{"clientId": "${client_id}", "connected": 1, "version": "1", "services": {"${sensor_name}": "pvinverter"}}'
  will_message:
    topic: device/${client_id}/Status
    payload: '{"clientId": "${client_id}", "connected": 0, "version": "1"}'
  on_json_message:
    topic: device/${client_id}/DBus
    then:
      - globals.set:
          id: vrm_portal_id
          value: !lambda |-
            return x["portalId"];
      - globals.set:
          id: sensor_id
          value: !lambda |-
            return x["deviceInstance"]["${sensor_name}"];
      - mqtt.publish_json:
          # Anzeigename in Venus OS
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/CustomName", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = "Solax X1 boost";
      - mqtt.publish_json:
          # 0=AC input 1; 1=AC output; 2=AC input 2
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Position", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 1;
      - mqtt.publish_json:
          # Nennleistung des Wechselrichters in W
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/MaxPower", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 2000;

and the corresponding sensor data:

`sensor:
  - platform: solax_x1_mini
    ac_power:
      name: "${name} ac power"
      id: inverter_active_power
      on_value:
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/Power", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_active_power).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L1/Power", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_active_power).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L2/Power", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L3/Power", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
    energy_today:
      name: "${name} energy today"
      id: inverter_daily_energy
      on_value:
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/Energy/Forward", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_daily_energy).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L1/Energy/Forward", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_daily_energy).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L2/Energy/Forward", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L3/Energy/Forward", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
    energy_total:
      name: "${name} energy total"
    dc1_voltage:
      name: "${name} dc1 voltage"
    dc2_voltage:
      name: "${name} dc2 voltage"
    dc1_current:
      name: "${name} dc1 current"
    dc2_current:
      name: "${name} dc2 current"
    ac_current:
      name: "${name} ac current"
      id: inverter_grid_current
      on_value:
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/Current", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_grid_current).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L1/Current", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_grid_current).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L2/Current", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L3/Current", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
    ac_voltage:
      name: "${name} ac voltage"
      id: inverter_grid_voltage
      on_value:
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/Voltage", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_grid_voltage).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L1/Voltage", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = id(inverter_grid_voltage).state;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L2/Voltage", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
      - mqtt.publish_json:
          topic: !lambda return str_sprintf("W/%s/pvinverter/%d/Ac/L3/Voltage", id(vrm_portal_id).c_str(), id(sensor_id));
          payload: |-
            root["value"] = 0;
    ac_frequency:
      name: "${name} ac frequency"
    temperature:
      name: "${name} temperature"
    runtime_total:
      name: "${name} runtime total"
    mode:
      name: "${name} mode"
    error_bits:
      name: "${name} error bits"
    grid_voltage_fault:
      name: "${name} grid voltage fault"
    grid_frequency_fault:
      name: "${name} grid frequency fault"
    dc_injection_fault:
      name: "${name} dc injection fault"
    temperature_fault:
      name: "${name} temperature fault"
    pv1_voltage_fault:
      name: "${name} pv1 voltage fault"
    pv2_voltage_fault:
      name: "${name} pv2 voltage fault"
    gfc_fault:
      name: "${name} gfc fault"`

I don't know if this is related to the driver or to the code I use. I can see the messages comming in with the MQTT Explorer. 

If this is not installation related, what does the status payload look like that you are publishing to the driver to register your device?

I'll update the issue with some screenshots tomorrow from the MQTT Explorer in Cerbo Firmware Version 3.14 and 3.21

**Comment by freakent** [comment]
After the VenusOS upgrade did you run setup.sh again? You really need to check the log file in var/log/dbus-mqtt-devices/current to check for errors. Please attach the relevant section of the log.

From 0.6.5 onwards, you no longer need to run setup.sh after a venusOS upgrade. I have just released v0.6.6 I suggest you upgrade to that then check your system again.

**Comment by UdoD5** [comment]
Hi, 
the log file looks clean, I updated to Version 0.6.6 with Firmware 3.14
 
root@einstein:~# tail -f /var/log/dbus-mqtt-devices/current
@4000000065dd9b2a154b02fc INFO:device_service_config:About to open config file
@4000000065dd9b2b38482e24 INFO:device_service:Registering service pvinverter for client PvInverter1 at path com.victronenergy.pvinverter.mqtt_PvInverter1_L1
@4000000065dd9b2c0b6169fc INFO:settingsdevice:Setting /Settings/Devices/mqtt_PvInverter1_L1/ClassAndVrmInstance does not exist yet or must be adjusted
@4000000065dd9b2c1710ea34 INFO:vedbus:registered ourselves on D-Bus as com.victronenergy.pvinverter.mqtt_PvInverter1_L1
@4000000065dd9b2c1939830c INFO:device_service:Registered Service com.victronenergy.pvinverter.mqtt_PvInverter1_L1 under DeviceInstance 2
@4000000065dd9b2c1978706c INFO:device_manager:publish {'portalId': 'xyz', 'deviceInstance': {'L1': 2}} to device/PvInverter1/DBus, status is 0
@4000000065dd9b2d05e4c03c INFO:device_service:value changed, path: /Position, value: 1
@4000000065dd9b2d07b5424c INFO:device_service:value changed, path: /Ac/MaxPower, value: 2000

The PV Inverter is visible with values, so basically everything works.

Cerbo GX Firmware Update to 3.22:

Solax PV Inverter does not show up.. 

Error log:

@4000000065dd9b2d05e4c03c INFO:device_service:value changed, path: /Position, value: 1
@4000000065dd9b2d07b5424c INFO:device_service:value changed, path: /Ac/MaxPower, value: 2000
@4000000065ddb1ee1dfbd1c4 *** CCGX booted (0) ***
@4000000065ddb30814f1714c *** starting dbus-mqtt-devices ***
@4000000065ddb30a24ab17b4 dbus_mqtt_devices v{} 0.6.6
@4000000065ddb30a24ab3adc INFO:logger:Loglevel set to INFO
@4000000065ddb30a24f474f4 INFO:dbus_mqtt_devices:-------- dbus_mqtt_devices, v0.6.6 is starting up --------
@4000000065ddb30a2943aebc INFO:device_manager:Using portalId xyz
@4000000065ddb30b25fa6b74 INFO:mqtt_gobject_bridge:[Init] Connecting to local broker
@4000000065ddb30b299e2eb4 INFO:device_manager:[Connected] Result code 0



<img width="761" alt="27-02-2024 11-07-55" src="https://github.com/freakent/dbus-mqtt-devices/assets/125830597/80aa89e6-9be8-48fd-8701-479f29d2a6fa">

**Comment by freakent** [comment]
Can you clarify what you mean by 

> Solax PV Inverter does not show up..

Does not show up how and where?

You do know that you should be looking under the N topics, right?

**Reply by UdoD5** [reply]
No I didn't know, 
the only PV inverters listed under N are the Hoymiles, the Solax is missing. It's also missing in the listing on the remote control. 

But i Just looked into /data/log/dbus-mqtt current:

There are a lot of errors:

@4000000065ddb1682813e534   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 356, in _handle_write
@4000000065ddb1682814102c     self._set_dbus_value(service, '/' + path, value)
@4000000065ddb16828141be4   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 571, in _set_dbus_value
@4000000065ddb1682814279c     return self._dbus_conn.call_blocking(service, path, None, 'SetValue', 'v', [value])
@4000000065ddb1682814373c   File "/usr/lib/python3.8/site-packages/dbus/connection.py", line 666, in call_blocking
@4000000065ddb168281690cc     reply_message = self.send_message_with_reply_and_block(
@4000000065ddb1682816a06c dbus.exceptions.DBusException: org.freedesktop.DBus.Error.UnknownObject: Method "SetValue" with signature "v" on interface "(null)" doesn't exist
@4000000065ddb1682816b7dc 
@4000000065ddb16d21539fb4 ERROR:dbus_mqtt:[Request] Error in request: W/48e7da881457/pvinverter/2/Ac/Voltage b'{"value":234.8}'
@4000000065ddb16d216ff984 Traceback (most recent call last):
@4000000065ddb16d21752d8c   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 312, in _on_message
@4000000065ddb16d217544fc     self._handle_write(topic, msg.payload)
@4000000065ddb16d21763b14   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 356, in _handle_write
@4000000065ddb16d21764ab4     self._set_dbus_value(service, '/' + path, value)
@4000000065ddb16d2176566c   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 571, in _set_dbus_value
@4000000065ddb16d21766224     return self._dbus_conn.call_blocking(service, path, None, 'SetValue', 'v', [value])
@4000000065ddb16d2176eadc   File "/usr/lib/python3.8/site-packages/dbus/connection.py", line 666, in call_blocking
@4000000065ddb16d2176fe64     reply_message = self.send_message_with_reply_and_block(
@4000000065ddb16d217930e4 dbus.exceptions.DBusException: org.freedesktop.DBus.Error.UnknownObject: Method "SetValue" with signature "v" on interface "(null)" doesn't exist
@4000000065ddb16d21794c3c 
@4000000065ddb17228bfdca4 ERROR:dbus_mqtt:[Request] Error in request: W/48e7da881457/pvinverter/2/Ac/Voltage b'{"value":234.8}'
@4000000065ddb17228d12e3c Traceback (most recent call last):
@4000000065ddb17228dbbd5c   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 312, in _on_message
@4000000065ddb17228dbdc9c     self._handle_write(topic, msg.payload)
@4000000065ddb17228dbe854   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 356, in _handle_write
@4000000065ddb17228dbf40c     self._set_dbus_value(service, '/' + path, value)
@4000000065ddb17228dbfbdc   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 571, in _set_dbus_value
@4000000065ddb17228dc0b7c     return self._dbus_conn.call_blocking(service, path, None, 'SetValue', 'v', [value])
@4000000065ddb17228dd6b0c   File "/usr/lib/python3.8/site-packages/dbus/connection.py", line 666, in call_blocking
@4000000065ddb17228dd7aac     reply_message = self.send_message_with_reply_and_block(
@4000000065ddb17228dd8664 dbus.exceptions.DBusException: org.freedesktop.DBus.Error.UnknownObject: Method "SetValue" with signature "v" on interface "(null)" doesn't exist
@4000000065ddb17228dd99ec 
@4000000065ddb17738838d5c ERROR:dbus_mqtt:[Request] Error in request: W/48e7da881457/pvinverter/2/Ac/Voltage b'{"value":234.8}'
@4000000065ddb1773883c024 Traceback (most recent call last):
@4000000065ddb1773883c7f4   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 312, in _on_message
@4000000065ddb1773883d3ac     self._handle_write(topic, msg.payload)
@4000000065ddb1773883df64   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 356, in _handle_write
@4000000065ddb1773883eb1c     self._set_dbus_value(service, '/' + path, value)
@4000000065ddb177388640c4   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 571, in _set_dbus_value
@4000000065ddb17738865064     return self._dbus_conn.call_blocking(service, path, None, 'SetValue', 'v', [value])
@4000000065ddb17738866004   File "/usr/lib/python3.8/site-packages/dbus/connection.py", line 666, in call_blocking
@4000000065ddb17738866fa4     reply_message = self.send_message_with_reply_and_block(
@4000000065ddb17738867b5c dbus.exceptions.DBusException: org.freedesktop.DBus.Error.UnknownObject: Method "SetValue" with signature "v" on interface "(null)" doesn't exist
@4000000065ddb1773888501c 
...

**Comment by freakent** [comment]
You must subscribe to the N/* topics to view values and you will also need to use a keepalive, please refer to Victron's documentation https://github.com/victronenergy/dbus-flashmq

Can you pipe the output of /var/log/dbus-mqtt/current through tai64nlocal so that we can see if they are recent?
tail -f /var/log/dbus-mqtt/current | tai64nlocal

Also when you move VenusOS to 3.20 it uses flashmq so you would need to check the log file in /var/log/flashmq instead of /var/log/dbus-mqtt. I do not have /var/log/dbus-mqtt on my system.

**Reply by UdoD5** [reply]
Hi, this is the output of /data/log/dbus-mqtt/current

root@einstein:~# tail -f /var/log/dbus-mqtt/current | tai64nlocal
2024-02-27 09:56:08.578911500   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 356, in _handle_write
2024-02-27 09:56:08.578916500     self._set_dbus_value(service, '/' + path, value)
2024-02-27 09:56:08.578918500   File "/opt/victronenergy/dbus-mqtt/dbus_mqtt.py", line 571, in _set_dbus_value
2024-02-27 09:56:08.578922500     return self._dbus_conn.call_blocking(service, path, None, 'SetValue', 'v', [value])
2024-02-27 09:56:08.578925500   File "/usr/lib/python3.8/site-packages/dbus/connection.py", line 666, in call_blocking
2024-02-27 09:56:08.578958500     reply_message = self.send_message_with_reply_and_block(
2024-02-27 09:56:08.579067500 dbus.exceptions.DBusException: org.freedesktop.DBus.Error.UnknownObject: Method "SetValue" with signature "v" on interface "(null)" doesn't exist
2024-02-27 09:56:08.579075500
2024-02-27 09:56:13.448122500 INFO:dbus_mqtt:[OwnerChange] Service disappeared: com.victronenergy.packageManager
2024-02-27 09:56:52.509681500 *** CCGX booted (0) ***

this is the output of /data/log/flashmq/current

@4000000065ddc4281623a404 [2024-02-27 11:14:38.365] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Energy/Forward': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Energy/Forward
@4000000065ddc4281623d6cc [2024-02-27 11:14:38.365] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Energy/Forward': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Energy/Forward
@4000000065ddc4281623f60c [2024-02-27 11:14:38.366] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Energy/Forward': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Energy/Forward
@4000000065ddc4281626670c [2024-02-27 11:14:38.366] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Current
@4000000065ddc4281626864c [2024-02-27 11:14:38.366] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Current
@4000000065ddc4281627034c [2024-02-27 11:14:38.367] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Current
@4000000065ddc4281627bae4 [2024-02-27 11:14:38.367] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Current
@4000000065ddc4281627d63c [2024-02-27 11:14:38.367] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Voltage
@4000000065ddc428162833fc [2024-02-27 11:14:38.367] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Voltage
@4000000065ddc42816284f54 [2024-02-27 11:14:38.367] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Voltage
@4000000065ddc4281628a544 [2024-02-27 11:14:38.368] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Voltage
...

**Reply by freakent** [reply]
Please pipe it thorough tai64nlocal so we can check the timestamps. It's make no sense at all that you would have both dbus-mqtt and flashmq on same system, unless you are running one of the 3.20 betas.

**Reply by UdoD5** [reply]
Hi the first log file was piped trough tai64nlocal. The second logfile does contain the timestamps already.  I guess the directory for the logfiles /data/log/dbus-mqtt/current is still available, it wasn't deleted when upgrading to 3.22. 
tail -f /var/log/flashmq/current | tai64nlocal is constantly refreshed (5sec)  

root@einstein:~# tail -f /var/log/flashmq/current | tai64nlocal
2024-02-27 11:55:13.210681500 [2024-02-27 11:55:13.210] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Current
2024-02-27 11:55:13.215489500 [2024-02-27 11:55:13.215] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Current
2024-02-27 11:55:13.230522500 [2024-02-27 11:55:13.230] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Voltage
2024-02-27 11:55:13.230770500 [2024-02-27 11:55:13.230] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Voltage
2024-02-27 11:55:13.231195500 [2024-02-27 11:55:13.230] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Voltage
2024-02-27 11:55:13.231423500 [2024-02-27 11:55:13.231] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Voltage
2024-02-27 11:55:13.248803500 [2024-02-27 11:55:13.248] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Power': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Power
2024-02-27 11:55:13.249065500 [2024-02-27 11:55:13.248] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Power': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Power
2024-02-27 11:55:13.257693500 [2024-02-27 11:55:13.256] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Power': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Power
2024-02-27 11:55:13.257706500 [2024-02-27 11:55:13.257] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Power': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Power
2024-02-27 11:55:18.179826500 [2024-02-27 11:55:18.179] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Energy/Forward': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Energy/Forward
2024-02-27 11:55:18.180164500 [2024-02-27 11:55:18.179] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Energy/Forward': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Energy/Forward
2024-02-27 11:55:18.188023500 [2024-02-27 11:55:18.187] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Energy/Forward': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Energy/Forward
2024-02-27 11:55:18.188309500 [2024-02-27 11:55:18.188] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Energy/Forward': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Energy/Forward
2024-02-27 11:55:18.213452500 [2024-02-27 11:55:18.213] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Current
2024-02-27 11:55:18.214382500 [2024-02-27 11:55:18.213] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Current
2024-02-27 11:55:18.214397500 [2024-02-27 11:55:18.213] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Current
2024-02-27 11:55:18.225195500 [2024-02-27 11:55:18.223] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Current': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Current
2024-02-27 11:55:18.225209500 [2024-02-27 11:55:18.224] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Voltage
2024-02-27 11:55:18.225216500 [2024-02-27 11:55:18.224] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L1/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L1/Voltage
2024-02-27 11:55:18.225324500 [2024-02-27 11:55:18.224] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L2/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L2/Voltage
2024-02-27 11:55:18.230031500 [2024-02-27 11:55:18.229] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/L3/Voltage': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/L3/Voltage
2024-02-27 11:55:18.247549500 [2024-02-27 11:55:18.247] [ERROR] Error in flashmq_plugin_acl_check when handling 'W/48e7da881457/pvinverter/2/Ac/Power': Can't find dbus service for W/48e7da881457/pvinverter/2/Ac/Power

**Reply by freakent** [reply]
I think it would help if you upgraded to the latest production release of VenusOS and latest dbus-mqtt-devices. Can you run Python on another machine? If so, run both the [combo-simulator.py](https://github.com/freakent/dbus-mqtt-devices/blob/main/samples/combo-simulator.py) and the [keepalive](https://github.com/freakent/dbus-mqtt-devices/blob/main/samples/keepalive.py). 

At the moment it looks like dbus-mqtt-devices is working as it should, these look like either coding issues or environment issues.

**Reply by UdoD5** [reply]
Hi Martin, 
thank you for looking into it. I agree that the code requires some update. Unfortunately I'm not a coder. So I have to wait until somebody updates the yaml file accordingly or supports me with the necessary changes (I posted in a different forum).  For now I went back to version 3.14
